### PR TITLE
Doc stuff

### PR DESCRIPTION
**Pull request by mcarton**

### CODE REVIEW

File: src/consts.rs [hunk 1]
```diff
@@ -30,7 +30,7 @@ impl From<FloatTy> for FloatWidth {
     }
 }
 
-/// a Lit_-like enum to fold constant `Expr`s into
+/// A `Lit_`-like enum to fold constant `Expr`s into.
```
Comment (llogiq) [inline src/consts.rs]: It's called `LitKind` these days, btw.

File: README.md [hunk 2]
```diff
@@ -42,6 +42,7 @@ name
 [cyclomatic_complexity](https://github.com/Manishearth/rust-clippy/wiki#cyclomatic_complexity)                       | warn    | finds functions that should be split up into multiple functions
 [deprecated_semver](https://github.com/Manishearth/rust-clippy/wiki#deprecated_semver)                               | warn    | `Warn` on `#[deprecated(since = "x")]` where x is not semver
 [derive_hash_xor_eq](https://github.com/Manishearth/rust-clippy/wiki#derive_hash_xor_eq)                             | warn    | deriving `Hash` but implementing `PartialEq` explicitly
+[doc_markdown](https://github.com/Manishearth/rust-clippy/wiki#doc_markdown)                                         | warn    | checks for the presence of the `_` character outside ticks in documentation
```
Comment (Manishearth) [inline README.md]: Perhaps "an odd number of `_`s"? It can be used to create italics.

**Comment by llogiq** [comment]
Yeah, I thought we may be able to check basic markdown validity. Also we could check camel-cased words outside of `\``s and lint if there is a type with that name.

File: README.md (hunk 2 above)
Comment (mcarton) [inline README.md]: Hum, did not know that, always used `*`s. It’ll be difficult to fix without a real markdown parser :s

Comment (mcarton) [inline README.md]: /me hates markdown.

**Comment by llogiq** [comment]
Shouldn't we be able to use whatever rustdoc uses?

**Comment by mcarton** [comment]
It looks like it uses hoedown (a C library!) to directly convert markdown to html.

**Comment by llogiq** [comment]
Ouch. Ok, thinking a bit more about it, we could at least lint underscores when between two word chars at least, because that usually creates some _really_ weird looking markup and is usually not what we want.

**Comment by mcarton** [comment]
Ok, I’ve improved it, it should not warn on `_`-based emphasis and now also warn on paths (something with `::`) outside ticks.

File: src/bit_mask.rs [hunk 3]
```diff
@@ -13,14 +13,14 @@ use utils::span_lint;
...
 ///
-/// |Comparison  |Bit-Op|Example     |is always|Formula               |
-/// |------------|------|------------|---------|----------------------|
-/// |`==` or `!=`| `&`  |`x & 2 == 3`|`false`  |`c & m != c`          |
-/// |`<`  or `>=`| `&`  |`x & 2 < 3` |`true`   |`m < c`               |
-/// |`>`  or `<=`| `&`  |`x & 1 > 1` |`false`  |`m <= c`              |
-/// |`==` or `!=`| `|`  |`x | 1 == 0`|`false`  |`c | m != c`          |
-/// |`<`  or `>=`| `|`  |`x | 1 < 1` |`false`  |`m >= c`              |
-/// |`<=` or `>` | `|`  |`x | 1 > 0` |`true`   |`m > c`               |
+/// |Comparison  |Bit Op |Example     |is always|Formula               |
```
Comment (llogiq) [inline src/bit_mask.rs]: Not: why add another whitespace here?

File: src/doc.rs [hunk 4]
```diff
@@ -0,0 +1,133 @@
+use rustc::lint::*;
+use std::borrow::Cow;
+use syntax::ast;
+use syntax::codemap::Span;
+use utils::span_lint;
+
+/// **What it does:** This lint checks for the presence of `_`, `::` or camel-case words outside
```
Comment (llogiq) [inline src/doc.rs]: There should be some docs about emphasis by underscores.

File: tests/compile-fail/doc.rs [hunk 5]
```diff
@@ -0,0 +1,30 @@
...
+/// ```
+fn multiline_ticks() {
+}
+
+/// The `main` function is the entry point of the program. Here it only calls the `foo_bar` and
+/// `multiline_ticks` functions.
+fn main() {
+    foo_bar();
+    multiline_ticks();
+}
```
Comment (llogiq) [inline tests/compile-fail/doc.rs]: I'd like to see a test for multiline emphasis.

**Comment by mcarton** [comment]
@llogiq comments addressed

**Comment by llogiq** [comment]
Thanks @mcarton, this is a good lint to have!

**Pull request merged**